addons/im_livechat/static/src/embed/common/thread_model_patch.js

```diff
@@ -47,4 +48,20 @@ patch(Thread.prototype, {
     get hasWelcomeMessage() {
         return this.channel_type === "livechat" && !this.chatbot && !this.requested_by_operator;
     },
+    /** @returns {Promise<import("models").Message} */
+    async post(body, params) {
+        if (
+            this.channel_type === "livechat" &&
+            this.store.env.services["im_livechat.livechat"].state !== SESSION_STATE.PERSISTED
```

Comment:
condition on persisted was added here by mistake?

Reply:
No this is intentional. The override was changing `thread` in the process of persist.
Since the `post` function is now owned by the thread, if the thread is not persisted it should go in that branching, otherwise it doesn't. This is so that when persist returns a thread, it doesn't try to `.persist()` again